google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/Fields.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2017 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+import com.google.api.services.bigquery.model.TableFieldSchema;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.Lists;
+import java.io.Serializable;
+import java.util.AbstractList;
+import java.util.List;
+import java.util.Map;
+
+public final class Fields extends AbstractList<Field> implements Serializable {
+
+  private static final long serialVersionUID = 8736258375638733316L;
+
+  private final List<Field> fields;
+  private final Map<String, Integer> nameIndex;
+
+  private Fields(Iterable<Field> fields) {
+    this.fields = ImmutableList.copyOf(fields);
+    ImmutableMap.Builder<String, Integer> nameIndexBuilder = ImmutableMap.builder();
+    int index = 0;
+    for (Field field : fields) {
+      nameIndexBuilder.put(field.getName(), index++);
```

Comment:
I much prefer the increment to be on its own line.

Reply:
Well, I like using inline incrementing (feels like "true programming" :), but ok, changed.